Encryption.md

```diff
@@ -128,6 +127,26 @@ unique nonce with a length of 12 bytes (96 bits). Notice: the NIST
 specification uses a term “IV” for what is called “nonce” in the Parquet encryption design.
 
 
+#### 4.1.4 Invocation limit
+According to the section 8.3 of the NIST SP 800-38D document, one key can be used for up to 
+2^32 encryption operations with RGB-based nonces in the AES GCM cipher. If this limit is exceeded, 
+the data encryption can be broken. The bulk of modules in a parquet file are page headers and data 
+pages. Therefore, one encryption key must not be applied to more than 2^31 (~2 billion) pages. In 
+parquet files encrypted with multiple keys (footer and column keys), the invocation limit is 
+applied to each key separately.
+
+
+Parquet is a local library, with each instance running in a process unaware of other instances. 
+Therefore Parquet cannot be responsible for enforcing the key invocation limits. The 
+responsibility lies with higher layers implementing the distributed applications that launch 
+cluster nodes and supply them with parquet encryption keys.
+
+Parquet implementations should implement a local invocation counter for each encryption key, and 
+throw an exception when the counter exceeds 2^32. While this does not enforce a system-wide limit, 
```

Comment:
Perhaps we can replace "throw an exception" with something more generic, such as "error out".

Reply:
Mm. Two things need to be done at that point 1) stop the writer so no more cipherblocks are produced/written 2) inform the user of the error. I'll update the text accordingly.

Reply:
"error out" implies that the operation stops, IMHO, but perhaps a better wording can summarize that idea.

Reply:
I agree that "throw an exception" implies to me a specific implementation  -- for example, in Rust there are no "exceptions" (though there are analgous concepts.

"Return an error and produce no more cipherblocks" seems like good phrasing to me

Reply:
SGTM